webcompat/webhooks/helpers.py

```diff
@@ -206,3 +230,177 @@ def new_opened_issue(payload):
+        response = private_issue_moderation(issue_info)
+        if response.status_code == 200:
+            return ('Moderated issue accepted',
+                    200, {'Content-Type': 'text/plain'})
+        else:
+            msg_log('private:moving to public failed', issue_number)
+            return ('ooops', 400, {'Content-Type': 'text/plain'})
+    elif (scope == 'private' and issue_info['state'] == 'closed'):
+        # private issue has been closed. It is rejected
+        # We need to patch with a template.
+        response = private_issue_rejected(issue_info)
+        if response.status_code == 200:
+            return ('Moderated issue rejected',
+                    200, {'Content-Type': 'text/plain'})
+        else:
+            msg_log('public rejection failed', issue_number)
+            return ('ooops', 400, {'Content-Type': 'text/plain'})
+    else:
+        return ('Not an interesting hook', 403, {'Content-Type': 'text/plain'})
+
+
+def repo_scope(source_repo):
+    """Check the scope nature of the repository.
+
+    The repository can be a
+    * known public.
+    * known private.
+    * or completely unknown.
+    """
+    scope = 'unknown'
+    if source_repo.endswith(PUBLIC_REPO.rsplit('/issues')[0]):
+        scope = 'public'
+    elif source_repo.endswith(PRIVATE_REPO.rsplit('/issues')[0]):
+        scope = 'private'
+    return scope
+
+
+def msg_log(msg, issue_number):
+    """Write a log with the reason and the issue number."""
+    log = app.logger
+    log.setLevel(logging.INFO)
+    msg = 'issue {issue} {msg}'.format(issue=issue_number, msg=msg)
+    log.info(msg)
+
+
+def prepare_accepted_issue(issue_info):
+    """Create the payload for the accepted moderated issue.
+
+    When the issue has been moderated as accepted,
+    we need to change a couple of things in the public space
+
+    - Title
+    - body
+    - Any labels from the private issue
+    """
+    # Extract the relevant information
+    public_url = issue_info['public_url']
+    body = issue_info['body']
+    title = issue_info['title']
+    issue_number = issue_info['number']
+    # Gets the labels from the body
+    labels = get_issue_labels(body)
+    labels.extend(issue_info['original_labels'])
+    # Let's remove action-needsmoderation in case it's here
+    if 'action-needsmoderation' in labels:
+        labels.remove('action-needsmoderation')
+    # Prepares the payload
+    payload_request = {
+        'labels': labels,
+        'title': title,
+        'body': body}
+    return payload_request
+
+
+def private_issue_moderation(issue_info):
+    """Write the private issue in public.
+
+    Send a GitHub PATCH to set labels and milestone for the issue.
+
+    PATCH /repos/:owner/:repo/issues/:number
+    {
+        "title": "a string for the title",
+        "body": "the full body",
+        "labels": ['Label1', 'Label2'],
+    }
+
+    Milestone should be already set on needstriage
```

Comment:
OK, it took my brain a second to get here, but my understand is:

1. anon public issue is filed, and set to needstriage w/ a label "needsmoderation", the body is just a placeholder.
2. the issue is moderated in private, by setting to 'accepted' milestone. 
3. all the data is gathered from the private issue, and PATCHed to the public issue, but we don't have to change the milestone, because it was already needstriage. but we remove the 'needsmoderation' label in `prepare_accepted_issue`.

Is my understanding correct?

Reply:
correct!